frontend/src/components/v2/UpgradeProjectAlert/UpgradeProjectAlert.tsx

```diff
@@ -0,0 +1,113 @@
+import { useCallback, useEffect, useState } from "react";
+import { useRouter } from "next/router";
+
+import { useNotificationContext } from "@app/components/context/Notifications/NotificationProvider";
+import { useProjectPermission } from "@app/context";
+import { useGetUpgradeProjectStatus, useUpgradeProject } from "@app/hooks/api";
+import { Workspace } from "@app/hooks/api/types";
+
+import { Alert } from "../Alert";
+import { Button } from "../Button";
+
+export type UpgradeProjectAlertProps = {
+  project: Workspace;
+};
+
+export const UpgradeProjectAlert = ({ project }: UpgradeProjectAlertProps): JSX.Element | null => {
+  const { createNotification } = useNotificationContext();
+  const router = useRouter();
+  const { membership } = useProjectPermission();
+  const upgradeProject = useUpgradeProject();
+  const [currentStatus, setCurrentStatus] = useState<string | null>(null);
+  const [isUpgrading, setIsUpgrading] = useState(false);
+  const {
+    data: projectStatus,
+    refetch: getLatestProjectStatus,
+    isLoading: statusIsLoading
+  } = useGetUpgradeProjectStatus(project.id);
+
+  const onUpgradeProject = useCallback(async () => {
+    if (upgradeProject.isLoading) {
+      return;
+    }
+    setIsUpgrading(true);
+    const PRIVATE_KEY = localStorage.getItem("PRIVATE_KEY");
+
+    if (!PRIVATE_KEY) {
+      createNotification({
+        type: "error",
+        text: "Private key not found"
+      });
+      return;
+    }
+
+    await upgradeProject.mutateAsync({
+      projectId: project.id,
+      privateKey: PRIVATE_KEY
+    });
+
+    await getLatestProjectStatus();
+
+    setTimeout(() => setIsUpgrading(false), 5_000);
+  }, []);
+
+  useEffect(() => {
```

Comment:
You can just interval feature in react-query. You don't need to manually do this. Based on the return api u can output it directly. You don't need to use another derived state